GLTF/src/GLTFAsset.cpp

```diff
@@ -440,6 +440,205 @@ GLTF::BufferView* packAccessorsForTargetByteStride(std::vector<GLTF::Accessor*>
 	return bufferView;
 }
 
+#ifdef USE_DRACO
+bool GLTF::Asset::compressPrimitives() {
+	int totalPrimitives = 0;
+	for (GLTF::Primitive* primitive : getAllPrimitives()) {
+		totalPrimitives++;
+		auto draco_ext_itr = primitive->extensions.find("KHR_draco_mesh_compression");
+		if (draco_ext_itr == primitive->extensions.end()) {
+			// No extension exists.
+			std::cout << "No extension exists for the primitive.\n";
+			continue;
+		}
+		GLTF::DracoExtension* draco_extension = (GLTF::DracoExtension*)draco_ext_itr->second;
+		draco::Mesh *draco_mesh = draco_extension->draco_mesh.get();
+		if (!draco_mesh) {
+			std::cout << "Duplicated mesh. Skipped.\n";
+			continue;
+		}
+
+		// Compress the mesh
+		// Setup encoder options.
+		draco::Encoder encoder;
+		int pos_quantization_bits= 14;
+		int tex_coords_quantization_bits = 10;
+		int normals_quantization_bits = 10;
+		int color_quantization_bits = 8;
+		// Used for compressing joint indices and joint weights.
+		int generic_quantization_bits = 8;
+
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION, pos_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD, tex_coords_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::NORMAL, normals_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::COLOR, color_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::GENERIC, generic_quantization_bits);
+
+		draco::EncoderBuffer buffer;
+		const draco::Status status = encoder.EncodeMeshToBuffer(*draco_mesh, &buffer);
+		if (!status.ok()) {
+			std::cerr << "Error: Encode mesh.\n";
+			return false;
+		}
+
+		// Add compressed data to bufferview
+		unsigned char* allocatedData = (unsigned char*)malloc(buffer.size());
+		std::memcpy(allocatedData, buffer.data(), buffer.size());
+		GLTF::BufferView* bufferView = new GLTF::BufferView(allocatedData, buffer.size());
+		draco_extension->bufferView = bufferView;
+		// Remove the mesh so duplicated primitives don't need to compress again.
+		draco_extension->draco_mesh.reset();
+	}
+	return true;
+}
+
+// TODO: Clean duplicated code from packAccessor()
```

Comment:
Not having to have the code separated out by macros should make this a lot easier